designer/restmodel/src/main/scala/pl/touk/nussknacker/restmodel/component/package.scala

```diff
@@ -16,6 +21,20 @@ package object component {
 
   type NodeId = String
 
+  @ConfiguredJsonCodec sealed trait NodeMetadata {
```

Comment:
Maybe let's call it `NodeUsageId`/`NodeUsageRef`/`NodeUsageData`
to be visible that it is related with usages mechanism. Node metadata looks like some generic metadata e.g. component reference, link documentation etc.